Mathlib/Algebra/Lie/DirectSum.lean

```diff
@@ -126,6 +126,28 @@ theorem bracket_apply (x y : ⨁ i, L i) (i : ι) : ⁅x, y⁆ i = ⁅x i, y i
   zipWith_apply _ _ x y i
 #align direct_sum.bracket_apply DirectSum.bracket_apply
 
+theorem lie_of_of_ne [DecidableEq ι] {i j : ι} (hij : j ≠ i) (x : L i) (y : L j) :
+    ⁅of L i x, of L j y⁆ = 0 := by
+  refine DFinsupp.ext fun k => ?_
+  rw [bracket_apply]
+  obtain rfl | hik := Decidable.eq_or_ne i k
+  · rw [of_eq_of_ne _ _ _ _ hij, lie_zero, zero_apply]
+  · rw [of_eq_of_ne _ _ _ _ hik, zero_lie, zero_apply]
+#align direct_sum.lie_of_of_ne DirectSum.lie_of_of_ne
+
+theorem lie_of_of_eq [DecidableEq ι] {i j : ι} (hij : j = i) (x : L i) (y : L j) :
+    ⁅of L i x, of L j y⁆ = of L i ⁅x, hij.recOn y⁆ :=
```

Comment:
This is a weird lemma (written by my less-experienced self, no doubt). Shouldn't it say:
```suggestion
theorem lie_of_same [DecidableEq ι] {i : ι} (x y : L i) :
    ⁅of L i x, of L i y⁆ = of L i ⁅x, y⁆ :=
```

Reply:
Done